### DISCUSSION TITLE
Explain WithTransactionScope

### DISCUSSION THREAD

**Discussion by yringler** [vkhorikov/CSharpFunctionalExtensions#544] [Q&A]
How does WithTransactionScope work? When should it be used?
There's a small example in the doc, but no explanation.

**Comment by vkhorikov** [comment]
Well, I personally never used that one.

**Comment by NormanMann** [comment]
It basically makes use of .net [TransactionScope](https://learn.microsoft.com/en-us/dotnet/api/system.transactions.transactionscope?view=net-9.0)

WithTransactionScope only really is applicable in those scenarios where you're already using .net TransactionScope
